src/tree.h

```diff
@@ -42,6 +42,8 @@ extern int git_tree_entry_icmp(const git_tree_entry *e1, const git_tree_entry *e
 void git_tree__free(void *tree);
 int git_tree__parse(void *tree, git_odb_object *obj);
 
+int git_treebuilder_write_with_buffer(git_oid *oid, git_treebuilder *bld, git_buf *tree);
```

Comment:
Sorry - one more minor nit.  For functions that are internal to the library, we use a double-underscore.  For example, this would be `git_treebuilder__write_with_buffer`.

But there's no need to put this in a header at all; what I would prefer to see is to revert the changes to `tree.h` and instead declare `write_with_buffer` as `static` in `tree.c`.  Would you be so kind?  Thanks again!